src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -364,6 +465,201 @@ private TerminalBasicInfo(TermInfo.Database db)
+            /// <summary>The format string for F12</summary>
+            public string KeyF12Format;
+            /// <summary>The format string for F13.</summary>
+            public string KeyF13Format;
+            /// <summary>The format string for F14.</summary>
+            public string KeyF14Format;
+            /// <summary>The format string for F15.</summary>
+            public string KeyF15Format;
+            /// <summary>The format string for F16</summary>
+            public string KeyF16Format;
+            /// <summary>The format string for F17.</summary>
+            public string KeyF17Format;
+            /// <summary>The format string for F18.</summary>
+            public string KeyF18Format;
+            /// <summary>The format string for F19.</summary>
+            public string KeyF19Format;
+            /// <summary>The format string for F20</summary>
+            public string KeyF20Format;
+            /// <summary>The format string for F21.</summary>
+            public string KeyF21Format;
+            /// <summary>The format string for F22.</summary>
+            public string KeyF22Format;
+            /// <summary>The format string for F23.</summary>
+            public string KeyF23Format;
+            /// <summary>The format string for F24</summary>
+            public string KeyF24Format;
+            /// <summary>The format string for KeyBackspace</summary>
+            public string KeyBackspaceFormat;
+            /// <summary>The format string for KeyClear</summary>
+            public string KeyClearFormat;
+            /// <summary>The format string for KeyDown</summary>
+            public string KeyDownFormat;
+            /// <summary>The format string for KeyUp</summary>
+            public string KeyUpFormat;
+            /// <summary>The format string for KeyHome</summary>
+            public string KeyHomeFormat;
+            /// <summary>The format string for KeyLeft</summary>
+            public string KeyLeftFormat;
+            /// <summary>The format string for KeyPageDown</summary>
+            public string KeyPageDownFormat;
+            /// <summary>The format string for KeyPageUp</summary>
+            public string KeyPageUpFormat;
+            /// <summary>The format string for KeyRight</summary>
+            public string KeyRightFormat;
+            /// <summary>The format string for KeyEnd</summary>
+            public string KeyEndFormat;
+            /// <summary>The format string for KeyEnter</summary>
+            public string KeyEnterFormat;
+            /// <summary>The format string for KeyHelp</summary>
+            public string KeyHelpFormat;
+            /// <summary>The format string for KeyPrint</summary>
+            public string KeyPrintFormat;
+            /// <summary>The format string for KeyInsert</summary>
+            public string KeyInsertFormat;
+            /// <summary>The format string for KeyDelete</summary>
+            public string KeyDeleteFormat;
```

Comment:
Why do we need all of these fields?  Seems like they're only being used in the TerminalKeyInfo ctor.  Could you just change your AddKey helper to be:
```C#
private void AddKey(TermInfo.Database db, int keyId, ConsoleKey key)
{
    string format = db.GetString(keyId);
    if (!string.IsNullOrEmpty(format))
    {
        KeyFormatToConsoleKey[format] = key;
    }
}
```
and then change the TerminalKeyInfo ctor to be something like:
```C#
private TerminalKeyInfo(TermInfo.Database db)
{
    KeyFormatToConsoleKey = new Dictionary<string, ConsoleKey);
    MaxKeyLength = MinKeyLength = 0;
    if (db != null)
    {
        AddKey(db, TermInfo.Database.KeyF1, ConsoleKey.F1);
        AddKey(db, TermInfo.Database.KeyF2, ConsoleKey.F2);
        ... // etc.
    }
}
```
and then get rid of all of those fields, all of the:
```
KeyF1Format = db != null ? db.GetString(TermInfo.Database.KeyF1) : string.Empty;
```
lines, etc.?